0_config/code/nldas_cube_utils.R

```diff
@@ -0,0 +1,57 @@
+
+nldas_cube_diff <- function(new_cube_df, old_cube_df_filename){
+  old_cube_df <- readRDS(old_cube_df_filename)
+  diffed_cells <- dplyr::anti_join(new_cube_df, old_cube_df,  by = c('x','y')) 
+  
+  if (any(is.na(diffed_cells))){
+    stop('found NA(s) in NLDAS cell diff. Check ', old_cube_df_filename, ' and `new_cube_df` data')
+  }
+  cube_bounds <- diffed_cells %>% 
+    summarize(x0 = min(x), x1 = max(x), y0 = min(y), y1 = max(y))
+  
+  return(cube_bounds)
+}
+
+
+calc_nldas_files <- function(time_range, time_stride, cube, ...){
+  
+  space_chunk <- sprintf("%s.%s_%s.%s", cube$x0, cube$x1, cube$y0, cube$y1)
+  variables <- c(...)
+  if (length(variables) == 0){
+    stop('must specify at least one variable as input to ... argument')
+  }
+  time_chunk_lead <- seq(time_range$t0, time_range$t1, by = time_stride)
```

Comment:
these are good variable names, but this would still be easier to read with a comment here and there stating your general intention with each cluster of code lines